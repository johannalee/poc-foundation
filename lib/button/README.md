# Button

```typescript
@Component({
  selector: 'poc-button',
  exportAs: 'pocButton',
  templateUrl: './button.html',
  styleUrls: ['./button.scss']
})
export class ButtonComponent {
  @Input()
  public classname = '';
  @Input()
  public placeholder = 'click me';
  @Input()
  public disabled = false;
}
```

## How to use 

```html
<poc-button
  classname="red"
  placeholder="Hello"
  [disabled]="true"
>
```
