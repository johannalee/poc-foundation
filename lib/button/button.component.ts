import {Component, Input} from '@angular/core';

@Component({
  selector: 'poc-button',
  exportAs: 'pocButton',
  templateUrl: './button.html',
  styleUrls: ['./button.scss'],
})
export class ButtonComponent {
  @Input()
  public classname = '';
  @Input()
  public placeholder = 'click me';
  @Input()
  public disabled = false;
}
