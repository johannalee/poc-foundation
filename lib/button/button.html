<button
  [class]="classname"
  [disabled]="disabled"
>
  {{ placeholder }}
</button>
