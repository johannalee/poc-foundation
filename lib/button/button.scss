@import url(https://fonts.googleapis.com/css?family=Poiret+One);

@mixin button-bg($bg) {
  background: $bg;
  &:hover {
    background:darken($bg,8%);
    transition: all 0.3s ease;
  }
  &:active {
    background:darken($bg,25%);
    outline: none;
    border: none;
  }
}

button {
  color:white;
  text-decoration:none;
  padding:5px 10px;
  border-radius:3px;
  font-family: 'Poiret One', cursive;
  font-size:2em;
}

.green {
  @include button-bg(#2ecc71);
}

.blue {
  @include button-bg(#3498db);
}

.yellow{
  @include button-bg(#f1c40f);
}

.red {
  @include button-bg(#e74c3c);
}

/*Other Stylez*/
.wrap {
  margin:0 auto;
  width:800px;
  text-align:center;
}

h1 {
  font-family: 'Poiret One', cursive;
  font-size:4em;
}

