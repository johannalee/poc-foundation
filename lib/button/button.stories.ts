import { storiesOf } from '@storybook/angular';
import { ButtonComponent } from './button.component';
import markdown from './README.md';

storiesOf('Button', module)
  .addParameters({
    jest: ['button']
  })
  .add(
    'with text',
    () => ({
      component: ButtonComponent,
      props: {
        placeholder: 'Hello Button',
        classname: 'red'
      }
    }),
    { notes: { markdown } }
  );
