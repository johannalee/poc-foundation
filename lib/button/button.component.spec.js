import runStoryshots from '@storybook/addon-storyshots';
import { imageSnapshot } from '@storybook/addon-storyshots-puppeteer';

runStoryshots({
  config: ({ configure }) => {
    configure(() => {
      require('./button.stories');
    }, module);
  },
  suite: 'ButtonComponent > DOM Snapshot'
});

runStoryshots({
  config: ({ configure }) => {
    configure(() => {
      require('./button.stories');
    }, module);
  },
  suite: 'ButtonComponent > Image Snapshot',
  test: imageSnapshot({ storybookUrl: 'http://localhost:6006/?path=/story/button--with-text' })
});
