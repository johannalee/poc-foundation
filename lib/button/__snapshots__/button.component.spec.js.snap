// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent > DOM Snapshot Button with text 1`] = `
<storybook-dynamic-app-root
  cfr={[Function CodegenComponentFactoryResolver]}
  data={[Function Object]}
  target={[Function ViewContainerRef_]}
>
  <poc-button>
    <button
      class="bc-red-light m-t-vvl"
    >
       Hello Button
  
    </button>
  </poc-button>
</storybook-dynamic-app-root>
`;
